Products/Zuul/routers/__init__.py

```diff
@@ -174,7 +174,14 @@ def asyncGetTree(self, id=None, additionalKeys=()):
             childData = Marshaller(child).marshal(keys)
```

Comment:
You could move this to an `else` below the `hasChildren` check to avoid marshalling when it's not needed.